File: src/main/java/com/goudourasv/domain/lectures/LecturesService.java
package com.goudourasv.domain.lectures;

import com.goudourasv.data.lectures.LecturesRepository;
import com.goudourasv.http.lectures.dto.LectureCreate;
import com.goudourasv.http.lectures.dto.LectureUpdate;

import javax.enterprise.context.ApplicationScoped;
import javax.transaction.Transactional;
import java.util.List;
import java.util.UUID;

@ApplicationScoped
public class LecturesService {
    private LecturesRepository lecturesRepository;

    public LecturesService(LecturesRepository lecturesRepository) {
        this.lecturesRepository = lecturesRepository;
    }

    @Transactional
    public List<Lecture> getFilteredLectures(UUID courseId) {
        List<Lecture> filteredLectures = lecturesRepository.getFilteredLectures(courseId);
        return filteredLectures;
    }

    @Transactional
    public Lecture getSpecificLecture(UUID lectureId) {
        Lecture lecture = lecturesRepository.getSpecificLecture(lectureId);
        return lecture;
    }

    @Transactional
    public Lecture createLecture(LectureCreate lectureCreate) {
        Lecture lecture = lecturesRepository.createLecture(lectureCreate);
        return lecture;
    }

    @Transactional
    public boolean deleteSpecificLecture(UUID id) {
        boolean deleted = lecturesRepository.deleteSpecificLecture(id);
        return deleted;
    }

    @Transactional
    public Lecture updateLecture(UUID id, LectureUpdate lectureUpdate) {
        Lecture lecture = lecturesRepository.replaceLecture(id, lectureUpdate);
        return lecture;
    }

    @Transactional
    public Lecture partiallyUpdateLecture(UUID id, LectureUpdate lectureUpdate) {
        Lecture lecture = lecturesRepository.partiallyUpdateLecture(id, lectureUpdate);
        return lecture;
    }
}

File: src/main/java/com/goudourasv/domain/lectures/LecturesService.kt
package com.goudourasv.domain.lectures

import com.goudourasv.data.lectures.LecturesRepository
import com.goudourasv.http.lectures.dto.LectureCreate
import com.goudourasv.http.lectures.dto.LectureUpdate
import java.util.*
import javax.enterprise.context.ApplicationScoped
import javax.transaction.Transactional

@ApplicationScoped
class LecturesService(private val lecturesRepository: LecturesRepository) {

    @Transactional
    fun getFilteredLectures(courseId: UUID): List<Lecture> {
        return lecturesRepository.getFilteredLectures(courseId)
    }

    @Transactional
    fun getSpecificLecture(lectureId: UUID): Lecture {
        return lecturesRepository.getSpecificLecture(lectureId)
    }

    @Transactional
    fun createLecture(lectureCreate: LectureCreate): Lecture {
        return lecturesRepository.createLecture(lectureCreate)
    }

    @Transactional
    fun deleteSpecificLecture(lectureId: UUID): Boolean {
        val deleted = lecturesRepository.deleteSpecificLecture(lectureId)
        return deleted
    }

    @Transactional
    fun replaceLecture(id: UUID, lectureCreate: LectureCreate): Lecture {
        return lecturesRepository.replaceLecture(id,lectureCreate)
    }

    @Transactional
    fun partiallyUpdateLecture(id: UUID, lectureUpdate: LectureUpdate): Lecture{
        return lecturesRepository.partiallyUpdateLecture(id,lectureUpdate)

    }
}